### ISSUE TITLE
Terraform Provider confused by removed application

### ISSUE THREAD

**Issue by spads-spads** [juju/terraform-provider-juju#231]
tl;dr: I tried to downgrade an application from one series to another, and hit two problems.  First, the provider would not redeploy downgraded versions for me.   Second, when I destroyed the application from within juju, the subsequent `terraform apply` caused a _segfault_ in the provider.

### Terraform Version

```
Terraform v1.5.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/juju/juju v0.6.0
```

### Affected Resource(s)

- juju_application

### Terraform Configuration Files

https://git.launchpad.net/canonical-terraform-plans/tree/entropy/environments/prod/main.tf

### Debug Output

https://pastebin.canonical.com/p/DWswmd5NDN/

### Expected Behaviour

A new application deployed to replace the old one.

### Actual Behaviour

The terraform provider referred to cached data instead of the actual current juju state, and segfaulted per the paste.

### Steps to Reproduce

1. `terraform apply`
2. Edit `main.tf` to replace `focal` with `bionic`
3. `juju remove-application squid-reverseproxy`
4. `terraform apply`

### Important Factoids

This is running in Prodstack 5, and as of 13 Jun 2023 is a WIP replacement service currently not carrying any production workload.

**Comment by juanmanuel-tirado** [comment]
Could you please confirm this still happens with the 0.8.0 version? Anyway, is this working with the juju CLI? I know Juju has some issues downgrading.

**Comment by spads-spads** [comment]
Ah, that seems to have gone far more smoothly!  

Juju doesn't support actual downgrading to my knowledge, so I expected the provider to work around that by reprovisioning somehow.  I may test further with 0.8.0 but this seems to have un-stuck me for now.
